Userland/Libraries/LibGUI/AbstractScrollableWidget.cpp

```diff
@@ -331,11 +331,13 @@ void AbstractScrollableWidget::set_automatic_scrolling_timer_active(bool active)
 Gfx::IntPoint AbstractScrollableWidget::automatic_scroll_delta_from_position(Gfx::IntPoint pos) const
 {
     Gfx::IntPoint delta { 0, 0 };
-
+    auto autoscroll_threshold_offset = -pos.y();
     if (pos.y() < m_autoscroll_threshold)
-        delta.set_y(clamp(-(m_autoscroll_threshold - pos.y()), -m_autoscroll_threshold, 0));
-    else if (pos.y() > widget_inner_rect().height() - m_autoscroll_threshold)
-        delta.set_y(clamp(m_autoscroll_threshold - (widget_inner_rect().height() - pos.y()), 0, m_autoscroll_threshold));
+        delta.set_y(clamp(-(m_autoscroll_threshold - pos.y()), -(m_autoscroll_threshold + autoscroll_threshold_offset), 0));
```

Comment:
In `-(m_autoscroll_threshold + autoscroll_threshold_offset)`, `autoscroll_threshold_offset` is equivalent to `-pos.y()` so you can also write this as `-(m_autoscroll_threshold - pos.y())` which is the same as the first argument to `clamp()`. Effectively this argument does not limit anything, so your entire expression is equivalent to `AK::min(pos.y() - m_autoscroll_threshold, 0)`

So this becomes:
```cpp
delta.set_y(AK::min(pos.y() - m_autoscroll_threshold, 0));
```